site/tests/test_record_file_permissions.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Zenodo is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Test Zenodo legacy permissions."""
+
+from io import BytesIO
+
+from invenio_rdm_records.proxies import current_rdm_records_service
+
+
+def create_record_w_file(client, record, headers):
+    """Create record with a file."""
+    # Create draft
+    record["files"] = {"enabled": True}
+    response = client.post("/records", json=record, headers=headers)
+    assert response.status_code == 201
+    recid = response.json["id"]
+
+    # Attach a file to it
+    response = client.post(
+        f"/records/{recid}/draft/files", headers=headers, json=[{"key": "test.pdf"}]
+    )
+    assert response.status_code == 201
+    response = client.put(
+        f"/records/{recid}/draft/files/test.pdf/content",
+        headers={
+            "content-type": "application/octet-stream",
+            "accept": "application/json",
+        },
+        data=BytesIO(b"testfile"),
+    )
+    assert response.status_code == 200
+    response = client.post(
+        f"/records/{recid}/draft/files/test.pdf/commit", headers=headers
+    )
+    assert response.status_code == 200
+
+    # Publish it
+    response = client.post(f"/records/{recid}/draft/actions/publish", headers=headers)
+    assert response.status_code == 202
+
+    return recid
```

Comment:
This looks like a copy of [this](https://github.com/inveniosoftware/invenio-rdm-records/blob/36a0abfb68a4a4e0ce2208087840de448a06c523/tests/resources/test_record_file_permissions.py#L23) from invenio-rdm-records, can we reuse it instead?

Reply:
I'm not sure i understand how.. unless we move it to `pytest-invenio`?

Reply:
Right, currently it doesn't seem that we can reuse utility functions from tests. We would probably need to group these utility functions per repo and make them available for others to be used. It looks like a bigger task, out of the scope of this task I would say.

Moving this to `pytest-invenio` doesn't seem possible because this is specific to invenio RDM.